File: src/main/java/dev/lampirg/consultationappointment/data/model/PersonRepository.java
package dev.lampirg.consultationappointment.data.model;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;

@NoRepositoryBean
public interface PersonRepository<T extends Person> extends JpaRepository<T, Integer> {
    T findByEmail(String email);
}

File: src/main/kotlin/dev/lampirg/consultationappointment/data/model/PersonRepository.kt
package dev.lampirg.consultationappointment.data.model

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.NoRepositoryBean

@NoRepositoryBean
interface PersonRepository<T : Person?> : JpaRepository<T, Int?> {
    fun findByEmail(email: String?): T
}